File: lib_local/gallery/src/main/java/com/yanzhenjie/durban/DurbanConfig.java
package com.yanzhenjie.durban;

import java.util.Locale;

/**
 * <p>Durban config.</p>
 * Created by Yan Zhenjie on 2017/5/30.
 */
public class DurbanConfig {
    private Locale mLocale;

    private DurbanConfig(Builder build) {
        this.mLocale = build.mLocale;
    }

    /**
     * Get {@link Locale}.
     *
     * @return {@link Locale}.
     */
    public Locale getLocale() {
        return mLocale;
    }

    /**
     * Create a new builder.
     */
    public static Builder newBuilder() {
        return new Builder();
    }

    public static final class Builder {
        private Locale mLocale;

        private Builder() {
        }

        /**
         * Set locale for language.
         *
         * @param locale {@link Locale}.
         * @return {@link Builder}.
         */
        public Builder setLocale(Locale locale) {
            this.mLocale = locale;
            return this;
        }

        /**
         * Create AlbumConfig.
         *
         * @return {@link DurbanConfig}.
         */
        public DurbanConfig build() {
            return new DurbanConfig(this);
        }
    }

}

File: lib_local/gallery/src/main/java/com/yanzhenjie/durban/DurbanConfig.kt
/*
 * Copyright © Yan Zhenjie. All Rights Reserved
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.yanzhenjie.durban

import java.util.Locale

/**
 * <p>Durban config.</p>
 * Created by Yan Zhenjie on 2017/5/30.
 */
class DurbanConfig(build: Builder) {
    var mLocale: Locale? = null
        private set

    init {
        this.mLocale = build.mLocale
    }

    companion object {
        /**
         * Create a new builder.
         */
        @JvmStatic
        fun newBuilder(): Builder {
            return Builder()
        }
    }

    class Builder internal constructor() {
        var mLocale: Locale? = null
            private set

        /**
         * Set locale for language.
         *
         * @param locale [Locale].
         * @return [Builder].
         */
        fun setLocale(locale: Locale): Builder {
            this.mLocale = locale
            return this
        }

        /**
         * Create AlbumConfig.
         *
         * @return [DurbanConfig].
         */
        fun build(): DurbanConfig {
            return DurbanConfig(this)
        }
    }

}